### ISSUE TITLE
Split window substitution makes split expand to fill screen.

### ISSUE THREAD

**Issue by ivegotasthma** [neovim/neovim#5725]
nvim version:
```
NVIM 0.1.7
Build type: RelWithDebInfo
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.1.7/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui      
For differences from Vim, see :help vim-differences

system vimrc file: "$VIM/sysinit.vim"
fall-back for $VIM: "/usr/share/nvim"
```
- Operating system/version:
Arch Linux  

- Terminal name/version:
lxterminal with gtk3 support  

- `$TERM`:
xterm-256color  

### Actual behaviour
substitution split buffer goes up and break other buffer

### Expected behaviour
not do that

### Steps to reproduce using `nvim -u NORC`
create a file with some text in it

```
nvim -u NONE $file-you-created
:set inccommand=split
:set laststatus=0
%s/some-text-in-file/asl;dkfjasl;fjkasl;fjaslfjaslfjasf
```

Here's a demo: https://vid.me/6wVL

**Comment by justinmk** [comment]
This appears to be the same as https://github.com/neovim/neovim/issues/5354.

https://github.com/neovim/neovim/pull/5370 is a fix, but it's waiting on tests and I'm not sure the fix there is really correct.

**Comment by justinmk** [comment]
Fixed by #5705
